core/src/test/scala/unit/kafka/log/LogManagerTest.scala

```diff
@@ -67,6 +67,26 @@ class LogManagerTest {
   @Test
   def testCreateLog() {
     val log = logManager.getOrCreateLog(new TopicPartition(name, 0), logConfig)
+    assertEquals(1, logManager.liveLogDirs.size)
+
+    val logFile = new File(logDir, name + "-0")
+    assertTrue(logFile.exists)
+    log.appendAsLeader(TestUtils.singletonRecords("test".getBytes()), leaderEpoch = 0)
+  }
+
+  /**
+   * Test that getOrCreateLog on a non-existent log creates a new log and that we can append to the new log.
+   * The LogManager is configured with one invalid log directory which should be marked as offline
+   */
+  @Test
+  def testCreateLogWithInvalidLogDir() {
+    val dirs = Seq(logDir, new File("\u0000"))
```

Comment:
Hmm, is this the behaviour we want? It seems to me that we should just fail start-up if log dirs is invalid (like this case). But we should handle the case where `getCanonicalPath` fails due to a bad disk better (which I think is the goal of this PR).

Reply:
@ijuma It seems like a naive test such and the Java doc says `returning false does not guarantee that the path is valid`. Not sure if it is worthwhile to 10 lines of code just to detect null character in the value of log directory config.

If we add this check in source code, I currently don't have a good idea to let getCanonicalPath() throw IOException. Do you like me to add this check in source code and then find another way to add test?

Reply:
OK, let's expand the comment about `getCanonicalPath` throwing an exception if the disk is bad to also mention that an invalid path throws the same error and we treat both the same at the moment. Also mention near where the File is created in the test that this triggers the same code path, but it's not the case we are trying to catch. It's just easier to simulate.

Reply:
Good point. Thanks for the suggestion. I have updated the comment as suggested.